### PR TITLE
Makes sharing with Snapchat work on Android

### PR DESCRIPTION
**Pull request by fearmear**
When I share through Snapchat on Android using latest version from master it just opens Snapchat and doesn't pass the file. This PR addresses the issue.

### CODE REVIEW

File: android/src/main/java/cl/json/social/EmailShare.java [hunk 1]
```diff
@@ -26,6 +26,11 @@ protected String getPackage() {
         return PACKAGE;
     }
 
+    @Override
+    protected String getComponentClass() {
+        return null;
```
Comment (mikehardy) [inline android/src/main/java/cl/json/social/EmailShare.java]: Any time a bunch of objects extend a parent object, and you need one thing in one of the children but none of the others, you should just provide a default implementation in the parent and override it in the one child that needs it

Alternatively, just override the one method you want to alter in the child (getPackage())

Touching all the other classes doesn't make sense to me

Comment (jgcmarins) [reply android/src/main/java/cl/json/social/EmailShare.java]: Agreed

Comment (fearmear) [reply android/src/main/java/cl/json/social/EmailShare.java]: Addressed your comments. Please check again.

**Comment by si0ls** [comment]
Any progress? 
I'm currently experiencing the same problem.

**Comment by mikehardy** [comment]
I think I’ve fixed the conflicts. Final comment on the PR is that it needs someone to actually test functionality. @si0ls install the branch (`npm i ‘git+https://github.com/fearmear/react-native-share.git#snapchat-android-fix'`) and verify it’s working and we can move forward I think?

**Comment by mikehardy** [comment]
Actually no - I see my conflict resolution did not seem to work - but it looks like a trival fix - just android imports (I did it in the github web ui so couldn’t check but it should be easy). You’ll need to resolve those to verify. You can start a review in github and using the little ‘+’ sign propose a suggested fix for acceptance to get the PR moving

**Comment by MateusAndrade** [comment]
@si0ls  you can test this please? Im without time on the last few days. 😢

**Comment by si0ls** [comment]
I confirm the build crash (https://i.imgur.com/TeWo8pM.png)

During this merge (https://github.com/fearmear/react-native-share/commit/fb1ff16364b0f0de299cdc60964e507ecbb959d5) the following changes were discarded:
- https://github.com/fearmear/react-native-share/commit/fb1ff16364b0f0de299cdc60964e507ecbb959d5#diff-ab7ecd9d2af3ade18f1e52953a3fe4bcL4
- https://github.com/fearmear/react-native-share/commit/fb1ff16364b0f0de299cdc60964e507ecbb959d5#diff-3fda3b361cc3bfe1acf44b82a68acc9dL167

Making these changes again fix the build issue, and Snapchat sharing works!

@fearmear May you fix this ?

**Comment by mikehardy** [comment]
Sorry I blew up the merge - last time I try to use the Github web UI for conflict resolution, it never seems to work out for me, and this is the worst result yet.

That was my fault, so I pulled the branch into my real dev environment, undid my busted merge, merged for real with a test that it build and ran the example, and force-pushed that back out to the PR branch

@si0ls I apologize for the hassle, could you double-check the result of that to make sure it still does what we think it does 🙏

**Comment by MateusAndrade** [comment]
> @si0ls I apologize for the hassle, could you double-check the result of that to make sure it still does what we think it does 🙏

This happens hahahah. I just checked and the check is running without any warning on the CI. I will try to test this today to ensure this is working. 😃

File: android/src/main/java/cl/json/social/ShareIntent.java [hunk 2]
```diff
@@ -195,6 +195,10 @@ public static boolean hasValidKey(String key, ReadableMap options) {
 
     protected abstract String getPackage();
 
+    protected String getComponentClass() {
+        return null;
```
Comment (jgcmarins) [inline android/src/main/java/cl/json/social/ShareIntent.java]: Is this right to return null here?

Comment (mikehardy) [reply android/src/main/java/cl/json/social/ShareIntent.java]: @jgcmarins yeah it's a default connected to the overrides in subclass overrides in response to previous review

**Comment by MateusAndrade** [comment]
@fearmear you can update your branch please? I give a quick look on your PR and it looks ok for me. =)

**Comment by StanislavMayorov** [comment]
> @fearmear you can update your branch please? I give a quick look on your PR and it looks ok for me. =)

@mikehardy Could you merge it please? I can rebase these commits to master. We have tested and used this version in production https://github.com/fearmear/react-native-share/commit/b878711abe3cc88c057b7bf22d9179754c69dccd

**Comment by mikehardy** [comment]
@StanislavMayorov I typically don't merge in this repo because I don't have time to support it (i.e., fix things when I break them) but I know I have reviewed this one before and it looked fine. I just updated the branch and it passed CI again so @jgcmarins and/or @MateusAndrade I think this one is good to go?

**Pull request merged**